### PR TITLE
[Pods] DCOS-9823: Configuration detail view

### PR DESCRIPTION
**Pull request by nLight**
This PR adds Configuration tab for pod.

screen shot 2016-09-20 at 14 44 58
screen shot 2016-09-20 at 14 27 56

### CODE REVIEW

File: src/js/components/PodContainerSpecView.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    return <DescriptionList hash={hash} />;
+  }
+
+  getLabelSection() {
+    let {container} = this.props;
+    if (!container.labels || !Object.keys(container.labels).length) {
+      return null;
+    }
+
+    console.log('Labels:', container.labels);
```
Comment (wavesoft) [inline src/js/components/PodContainerSpecView.js]: Stray console log

Comment (wavesoft) [inline src/js/components/PodContainerSpecView.js]: Stray console.log

File: src/js/components/PodContainerSpecView.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    let portConfigurations = MarathonConfigUtil.getPortDefinitionGroups(
+        id, endpoints, function (content, linkTo) {
+          return <a href={linkTo} target="_blank">{content}</a>;
+        }
+      ).map(function ({hash, headline}, index) {
+        return (
+          <DescriptionList className="nested-description-list"
+            hash={hash}
```
Comment (wavesoft) [inline src/js/components/PodContainerSpecView.js]: Can we create somewhere a `prune` function that removes keys with `null` or `undefined` values? Just to visually clean-up the empty labels:

image

**Comment by orlandohohmeier** [comment]
Wonder if we should apply the new nested list styling to the "service" (app/framework) configuration view as well... 

/cc @leemunroe

File: src/js/components/PodSpecView.js [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+    if (Object.keys(environment).length === 0) {
+      return null;
+    }
+
+    Object.keys(environment).forEach(function (key) {
+      let value = environment[key];
+      if (typeof value === 'object' && !Array.isArray(value) &&
+          value !== null) {
+        environment[key] = value.secret;
```
Comment (wavesoft) [inline src/js/components/PodSpecView.js]: Let's put a `Key Icon + Referred key name` instead of just the reference to the secret

**Comment by wavesoft** [comment]
Other than the few changes we mentioned, looks good to me! Great work! ✨  💯

**Comment by orlandohohmeier** [comment]
@nLight Could you please split this into smaller PRs? We try to keep our PRs to ~500 LOC max. which makes it easier to review the changes.

File: src/styles/components/pod-configuration-tab.less [hunk 4]
```diff
@@ -0,0 +1,12 @@
+
+.pod-config-container-group {
+    margin-left: 8px;
+    border-left: solid 2px #9351E5;
+    padding-left: 12px;
+}
+
+.pod-config-network-group {
```
Comment (wavesoft) [inline src/styles/components/pod-configuration-tab.less]: Perhaps we should break this into two different classes:

* `pod-config-container-group` - That contains the general rules (margin, padding, border style)
* `pod-config-container-group-network` - That adds the network color
* `pod-config-container-group-container ` - That adds the container color

**Comment by leemunroe** [comment]
@nLight @orlandohohmeier I love the nested styling. Makes it easier to follow. I think the colors need a bit more thought. The colors could be misinterpreted here, or maybe draw too much attention. How about using a more consistent ("boring") color?

image

I also like the icons. Are containers and networks the only sections we're using colors and icons for? What about health checks?

**Comment by nLight** [comment]
Hey @leemunroe, 
I generally like the boring color idea! We though we could use different colors to distinguish between sections. Right now containers and networks are heavy sections as they contain arrays of entities. We are open to any visual improvement suggestions /cc @wavesoft

**Comment by nLight** [comment]
Closing this PR, going to split it into 3-4 PRs.

**Comment by leemunroe** [comment]
@nLight Cool, yeh once we introduce other colors I'm concerned they could be misinterpreted as colors in DC/OS generally have meaning e.g. healthy, unhealthy. Neutral is a safer bet.

**Pull request closed without merging**